provider.go

```diff
@@ -0,0 +1,371 @@
+// Package libdnstemplate implements a DNS record management client compatible
+// with the libdns interfaces for Porkbun.
+package libdns_porkbun
```

Comment:
In the final publication, call the package just `porkbun` (or whatever you want, but `libdns_` is not a necessary / conventional thing).